src/backlight/backlight.c

```diff
@@ -361,26 +368,36 @@ static int clamp_brightness(sd_device *device, bool saved, unsigned max_brightne
         return 0;
 }
 
-static bool shall_clamp(sd_device *d) {
+static bool shall_clamp(sd_device *d, unsigned *ret) {
```

Comment:
maybe just make this return the percentage as `int`, and return 0 if we shall not clamp?

(and maybre rename to shall_clamp_percent()?

Reply:
I initially thought so, but people may want to set `ID_BACKLIGHT_CLAMP=0%`, to set the minimum brightness 1. To support such, we need to return `true` with `*ret = 0`. So, I'd like to keep the code as is.